src/Shared/LinkerFlags.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Internal
+{
+    internal static class LinkerFlags
+    {
+        /// <summary>
+        /// Flags for a member that is JSON (de)serialized.
+        /// </summary>
+        public const DynamicallyAccessedMemberTypes JsonSerialized = DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.PublicProperties;
```

Comment:
These don't apply recursively. So if the type in question is something like:

```C#
class Email { public Attachment Attachment {get; set;} }
class Attachment { public Name {get; set;} }
```

Annotating `Email` with this, will not preserve `Attachment.Name` which the serializer will try to access.
Currently there's no annotation which can be added for serialization purposes. For `System.Text.Json` we basically hope that source generators will solve this.

Reply:
That makes sense. These annotations mirror the JsonSerializer's current annotations. Once we have a AOT support there, we'll make sure these APIS are correctly set up to work with the source generator.